modules/upnp/doc_classes/UPNP.xml

```diff
@@ -16,6 +16,32 @@
 		[codeblock]
 		upnp.delete_port_mapping(port)
 		[/codeblock]
+		[b]Note:[/b] UPnP discovery blocks the main thread. To perform non-blocking discovery, use [Thread]s like this:
+		[codeblock]
+		# Replace this with your own server port number between 1025 and 65535.
+		const SERVER_PORT = 3928
+		var thread = null
+
+		func _upnp_setup(server_port):
+		    # UPNP queries take some time.
+		    var upnp = UPNP.new()
+		    var err = upnp.discover()
+
+		    if err:
+		        return push_error(err)
```

Comment:
`push_error` returns void.
`push_error` also seems to fail (causing the script to crash) when passed a non-string argument.

```gdscript
	if err != OK:
		push_error(str(err))
		return
```